lib/mm.rb

```diff
@@ -0,0 +1,78 @@
+require_relative 'mm_code'
+require_relative 'mm_outputs'
+require_relative 'mm_input'
+require_relative 'mm_gen_list'
+
+module Mastermind
+    def self.play(blank_list = [], new_list = GeneratedList.new, new_code = Code.new(new_list.random_code), is_new = true)
+        previous_guesses = blank_list
+        remaining_guesses = 10
+        pins_out = blank_list
+        gen_list = new_list
+        game_code = new_code
+        game_out = Outputs.new
+        first_launch = is_new
+        until game_code.matched do
+            system "clear"
+            if first_launch
+                game_out.welcome
+                first_launch = false
+            end
+            if remaining_guesses.zero?
+                puts "You ran out of guesses..."
```

Comment:
I like that you're thinking about dependency injection here but there seems to be a lot of default values in place. Could this be a place where you could use [composition](https://learnrubythehardway.org/book/ex44.html) in a constructor? With your defaults, you can at this point call `Mastermind.play` with no arguments which I think is a great choice. However, I find it easier to use composition because it's clearer to other developers what this class depends on (even if there are defaults).